src/main/java/seedu/address/storage/JsonAdaptedLesson.java

```diff
@@ -0,0 +1,106 @@
+
+    private final String code;
+    private final String type;
+    private final String date;
+    private final String location;
+    private final String weightage;
+
+    /**
+     * Constructs a {@code JsonAdaptedLesson} with the given lesson details.
+     */
+    @JsonCreator
+    public JsonAdaptedLesson(@JsonProperty("code") String code, @JsonProperty("type") String type,
+                             @JsonProperty("date") String date, @JsonProperty("location") String location,
+                             @JsonProperty("weightage") String weightage) {
+        this.code = code;
+        this.type = type;
+        this.date = date;
+        this.location = location;
+        this.weightage = weightage;
+    }
+
+    /**
+     * Converts a given {@code Lesson} into this class for Jackson use.
+     */
+    public JsonAdaptedLesson(Lesson source) {
+        code = source.getCode().code;
+        type = source.getTypeStr();
+        date = source.getDate().format(Task.FORMATTER);
+        location = source.getLocation().value;
+        weightage = Double.toString(source.getWeightage());
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted contact object into the model's {@code Lesson} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted contact.
+     */
+    public Lesson toModelType() throws IllegalValueException {
+
+        if (code == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Code.class.getSimpleName()));
+        }
+        if (!Code.isValidCode(code)) {
+            throw new IllegalValueException(Code.MESSAGE_CONSTRAINTS);
+        }
+        final Code modelCode = new Code(code);
+
+        if (type == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Type.class.getSimpleName()));
+        }
+        final Type modelType = Enum.valueOf(Type.class, type);
+
+        if (date == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                LocalDate.class.getSimpleName()));
+        }
+        try {
+            LocalDate.parse(date, Task.FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new IllegalValueException(Task.DATE_MESSAGE_CONSTRAINTS);
+        }
+        final LocalDate modelTime = LocalDate.parse(date, Task.FORMATTER);
+
+        if (location == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Email.class.getSimpleName()));
+        }
+        if (!Address.isValidAddress(location)) {
+            throw new IllegalValueException(Address.MESSAGE_CONSTRAINTS);
+        }
+        final Address modelLocation = new Address(location);
+
+        if (weightage == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Double.class.getSimpleName()));
```

Comment:
Not Double.class.getSimpleName() right? It should be Weightage

Reply:
There is no Weightage class. Weightage is stored as a double thats why I used Double class.

Reply:
Yeah but for the message you can't output like "missing a double", better to just replace it by "Weightage" (a String) since the Double.class.getName() is basically get the name of that class

Reply:
Yes ok I will change it